common/src/main/java/io/netty/util/internal/PlatformDependent0.java

```diff
@@ -311,6 +318,37 @@ static void copyMemory(Object src, long srcOffset, Object dst, long dstOffset, l
         }
     }
 
+    static boolean equals(final byte[] bytes1, final byte[] bytes2) {
+        if (bytes1.length != bytes2.length) {
+            return false;
+        }
+        if (bytes1.length == 0) {
+            return true;
+        }
+        int remainingBytes = bytes1.length & 7;
+        for (int i = bytes1.length - 8; i >= remainingBytes; i -= 8) {
+            final long word1 = UNSAFE.getLong(bytes1, ARRAY_BASE_OFFSET + i);
```

Comment:
@Scottmitch I think it's a lot easier to read with the local variables

> to communicate conceptual meaning of values with local variable then can we use a comment

I disagree sooo much :-) Don't use comments when you can just name your variable the right way...

> BTW 169 lines of byte code with inline suggestions vs 192 as is without any changes.

is this after javac? If you are thinking towards inlining, AFAIK what counts is the output after the HotSpot compiler frontend has processed the byte code.

Anyways, I removed the local variables, this is too small of a change to invest much time discussing :-).